bin/clang/astVisitor.cc

```diff
@@ -2697,7 +2703,10 @@ bool
 FirstPassASTVisitor::TraverseFunctionDecl(FunctionDecl *fd, DataRecursionQueue* /*queue*/)
 {
   PushGuard<FunctionDecl*> pg(CompilerGlobals::astContextLists.enclosingFunctionDecls, fd);
-  Parent::TraverseFunctionDecl(fd);
+  PragmaActivateGuard pag(fd, this, true/*always do first pass pragmas*/);
+  if (fd->isThisDeclarationADefinition()){
+    Parent::TraverseFunctionDecl(fd);
```

Comment:
Quick question did you mean to ignore the return value of the parent?

Reply:
I fixed this in the few places where it was easy to fix. We'll need a separate PR.